flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/cleanup/DefaultResourceCleaner.java

```diff
@@ -97,8 +104,8 @@
 
         private final RetryStrategy retryStrategy;
 
-        private final Collection<T> prioritizedCleanup = new ArrayList<>();
-        private final Collection<T> regularCleanup = new ArrayList<>();
+        private final LinkedHashMap<String, T> prioritizedCleanup = new LinkedHashMap<>();
+        private final Map<String, T> regularCleanup = new HashMap<>();
```

Comment:
Why not encapsulate the label and T in a pojo?

Reply:
good point - that feels to be more consistent with the actual purpose of that collection.